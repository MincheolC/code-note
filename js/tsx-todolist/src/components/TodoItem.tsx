import * as React from 'react';

interface Props {
    text: string;
    done: boolean;
    onToggle(): void;
    onRemove(): void;
}

// SFC StatelessFunctionComponent
const TodoItem: React.SFC<Props> = ({text, done, onToggle, onRemove}) => (
    <li>
        <b
            onClick={onToggle}
            style={{
                textDecoration: done ? 'line-through' : 'none'
            }}
        >
        {text}
        </b>
        <button onClick={onRemove} style={{all: 'unset' , marginLeft: '0.5rem'}}>[지우기]</button>
    </li>
);

export default TodoItem;