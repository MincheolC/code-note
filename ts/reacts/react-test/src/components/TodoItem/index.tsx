import React from 'react'

interface TodoItemProps {
  id: number,
  todo: string,
  onRemove: (id: number) => void,
}

const TodoItem: React.FC<TodoItemProps> = ({ id, todo, onRemove }) => {
  return (
    <li>
      <span>{todo}</span>
      <button onClick={() => onRemove(id)}>x</button>
    </li>
  )
}

export default TodoItem