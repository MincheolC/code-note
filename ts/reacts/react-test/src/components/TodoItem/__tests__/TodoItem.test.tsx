import React from 'react';
import renderer, { ReactTestRenderer, ReactTestRendererJSON } from 'react-test-renderer';
import TodoItem from '../';

const { act } = renderer;

test('Todo render properly', () => {
  const todo = { id: 1, todo: 'test', onRemove: () => {} };
  const component: ReactTestRenderer = renderer.create(
    <TodoItem id={todo.id} todo={todo.todo} onRemove={todo.onRemove}/>,
  );

  let tree = component.toJSON() as ReactTestRendererJSON;
  expect(tree).toMatchSnapshot();
});