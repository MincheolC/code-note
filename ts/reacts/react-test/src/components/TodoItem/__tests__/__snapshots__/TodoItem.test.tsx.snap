// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Todo render properly 1`] = `
<li>
  <span>
    test
  </span>
  <button
    onClick={[Function]}
  >
    x
  </button>
</li>
`;
